src/Sulu/Component/Content/Document/Subscriber/PHPCR/SuluNode.php

```diff
@@ -0,0 +1,477 @@
+
+    /**
+     * @param NodeInterface $node
+     */
+    public function __construct(NodeInterface $node)
+    {
+        $this->node = $node;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPath()
+    {
+        return $this->node->getPath();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return $this->node->getName();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAncestor($depth)
+    {
+        return $this->node->getAncestor($depth);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getParent()
+    {
+        return $this->node->getParent();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDepth()
+    {
+        return $this->node->getDepth();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSession()
+    {
+        return $this->node->getSession();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isNode()
+    {
+        return $this->node->isNode();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isNew()
+    {
+        return $this->node->isNew();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isModified()
+    {
+        return $this->node->isModified();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isSame(ItemInterface $otherItem)
+    {
+        return $this->node->isSame($otherItem);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function accept(ItemVisitorInterface $visitor)
+    {
+        return $this->node->accept($visitor);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function revert()
+    {
+        return $this->node->revert();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function remove()
+    {
+        return $this->node->remove();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addNode($relPath, $primaryNodeTypeName = null)
+    {
+        return $this->node->addNode($relPath, $primaryNodeTypeName);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addNodeAutoNamed($nameHint = null, $primaryNodeTypeName = null)
+    {
+        return $this->node->addNodeAutoNamed($nameHint, $primaryNodeTypeName);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function orderBefore($srcChildRelPath, $destChildRelPath)
+    {
+        return $this->node->orderBefore($srcChildRelPath, $destChildRelPath);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function rename($newName)
+    {
+        return $this->node->rename($newName);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setProperty($name, $value, $type = PropertyType::UNDEFINED)
+    {
+        $oldValue = $this->getPropertyValueWithDefault($name, null);
+        if ($oldValue !== null && gettype($value) !== gettype($oldValue)) {
+            $this->node->getProperty($name)->remove();
```

Comment:
The block implements the remove method a little bit different, but I think this difference doesn't apply in this case.